### ISSUE TITLE
Can't use init = 0 when >2 group specific terms in stan_{g}lmer

### ISSUE THREAD

**Issue by sambrilleman** [stan-dev/rstanarm#230]
#### Summary:
stan_{g}lmer always returns an error when init = 0 and there are > 2 group-specific parameters.

#### Description:
This is only a small bug and you may already be aware of it. Using init = "random" solves the problem. The bug seems to be caused by the Stan make_theta_L function. When init = 0 and there are > 2 group-specific parameters, then make_theta_L returns a vector which contains some NaN values at initialisation.

#### Reproducible Steps:
library(lme4)
library(rstanarm)
m1 <- stan_lmer(Reaction ~ poly(Days, 2) + (poly(Days, 2) | Subject), sleepstudy, init = 0)

And I seemed to get a similar error on other example datasets.

#### RStanARM Version:
2.15.3, downloaded from CRAN

#### R Version:
R 3.4.1

#### Operating System:
Windows 10

**Comment by bgoodri** [comment]
This is true, but unavoidable. I think the only thing we can do is document that people should not be doing `init=0` when there are correlation matrices.
